### ISSUE TITLE
[RESEARCH] LCX softmax temperature tau sweep: does tau != 1.0 improve slot retrieval quality?

### ISSUE THREAD

**Issue by Kenessy** [VRAXION/VRAXION#93]
## Origin
Pulse external review session -- 2026-02-18 (Pulse #22: Soft Top-K for Cooperative Gradients)

## Background
LCX slot retrieval already uses soft top-k: `topk()` selection followed by `softmax weighted sum` (swarm_model.py lines 2337-2338). This is the architecture Pulse proposed -- it's already implemented.

The missing piece: no temperature parameter on the LCX top-k softmax. Current code:

    weights = torch.nn.functional.softmax(topk_scores, dim=-1)  # no tau

Adding tau gives a tunable knob:
- Low tau (0.1-0.3): sharpens weights toward hard top-k (winner-takes-all within the K slots)
- tau=1.0: standard softmax (current behavior)
- High tau (2.0+): approaches uniform blend across all K slots

## Question
Is there an optimal LCX softmax temperature tau that improves Score Margin growth rate and bit_acc beyond the current tau=1.0 (implicit)?

## Relationship to GH #91 (top_k sweep)
These two are companion experiments on the same LCX retrieval path:
- GH #91: fix tau=1.0, vary top_k in {1,2,4,6,10}
- GH #93 (this): fix top_k=6, vary tau in [0.1, 2.0]

Run #91 first to establish optimal top_k, then sweep tau at that top_k.

## Implementation
Add `lcx_tau` parameter to SwarmByteRingModel config (default=1.0 = current behavior, backward compatible):

    # swarm_model.py ~line 2338 (non-bucketed path)
    weights = torch.nn.functional.softmax(topk_scores / self.lcx_tau, dim=-1)

    # swarm_model.py ~line 2376 (bucketed path)
    weights = torch.nn.functional.softmax(topk_scores / self.lcx_tau, dim=-1)

Two lines changed. Default lcx_tau=1.0 preserves current behavior exactly.

## Proposed Protocol -- Golden-Section Search

```python
import math

def golden_minimize(f, a, b, tol=0.05, max_iter=15):
    phi = (1 + math.sqrt(5)) / 2
    r = 1 / phi
    c = b - r * (b - a)
    d = a + r * (b - a)
    fc, fd = f(c), f(d)
    it = 0
    while (b - a) > tol and it < max_iter:
        if fc < fd:
            b, d, fd = d, c, fc
            c = b - r * (b - a)
            fc = f(c)
        else:
            a, c, fc = c, d, fd
            d = a + r * (b - a)
            fd = f(d)
        it += 1
    return (a + b) / 2

def f(tau):
    # train D=128 mini model, 500 steps, copy_echo256, seed=42, top_k=<best from #91>
    # return -bit_acc_median_100
    ...

best_tau = golden_minimize(f, a=0.1, b=2.0)
```

Starting bracket: a=0.1, b=2.0. First probes: c~0.86, d~1.24. Current behavior (tau=1.0) is the first interior probe.

## Metrics per eval
- bit_acc at step 200 and 500 (primary)
- Score Margin trajectory (routing decisiveness)
- lcx_clarity_index = (max_w - median_w) / max_w over top-k weights (new metric, see #86 Part 2 update)
- had_nan / had_inf / diverged (#86 Part 6)

## Note on Pulse framing
Pulse #22 proposed soft top-k for expert gating in a Prismion/Brainstem architecture with Fibonacci ant tiers -- all inapplicable. The underlying math (truncated exponential softmax over top-K) is exactly what LCX already does. The actual gap is just the missing tau parameter. Clarity index and tau sweep are the extractable elements.

## Related
- Code: swarm_model.py lines 2337-2338 (non-bucketed), 2375-2376 (bucketed)
- GH #91: top_k sweep (run first -- fix top_k before sweeping tau)
- GH #86 Part 2: lcx_clarity_index (new metric, complement to Score Margin)
- GH #87: golden_minimize() harness (reuse)
- Config: `lcx_top_k=6` in run_goldilocks.bat
